api/src/test/java/org/openmrs/ObsTest.java

```diff
@@ -840,41 +907,70 @@ public void removeGroupMember_shouldReturnFalseWhenANonExistentObsIsRemoved() th
 	
 	/**
 	 * @see Obs#removeGroupMember(Obs)
-	 * @verifies return true when an obs is removed
+	 * @verifies return true when an existing obs is removed from the group
 	 */
 	@Test
 	public void removeGroupMember_shouldReturnTrueWhenAnObsIsRemoved() throws Exception {
-		Obs obs = new Obs();
+		Obs obs = new Obs(2);
 		Obs member = new Obs();
 		obs.addGroupMember(member);
 		assertTrue(obs.isDirty());
 		resetObs(obs);
 		obs.removeGroupMember(member);
 		assertTrue(obs.isDirty());
 	}
+
+	/**
+	 * @see Obs#removeGroupMember(Obs)
+	 * @verifies return false when an new obs is removed from the group
+	 */
+	@Test
+	public void removeGroupMember_shouldReturnFalseForDirtyFlagWhenAnObsIsRemovedFromGroup() throws Exception {
+		Obs obs = new Obs();
+		Obs member = new Obs();
+		obs.addGroupMember(member);
+		assertFalse(obs.isDirty());
+		resetObs(obs);
+		obs.removeGroupMember(member);
+		assertFalse(obs.isDirty());
+	}
 	
 	/**
 	 * @see Obs#setGroupMembers(Set)
-	 * @verifies mark the obs as dirty when the set is changed from null to a non empty one
+	 * @verifies mark the existing obs as dirty when the set is changed from null to a non empty one
 	 */
 	@Test
-	public void setGroupMembers_shouldMarkTheObsAsDirtyWhenTheSetIsChangedFromNullToANonEmptyOne() throws Exception {
-		Obs obs = new Obs();
+	public void setGroupMembers_shouldMarkTheExistingObsAsDirtyWhenTheSetIsChangedFromNullToANonEmptyOne() throws Exception {
+		Obs obs = new Obs(5);
 		assertNull(Obs.class.getDeclaredField("groupMembers").get(obs));
 		Set members = new HashSet<>();
 		members.add(new Obs());
 		obs.setGroupMembers(members);
 		assertTrue(obs.isDirty());
 	}
-	
+
 	/**
 	 * @see Obs#setGroupMembers(Set)
-	 * @verifies mark the obs as dirty when the set is replaced with another with different members
+	 * @verifies do not mark the new obs as dirty when the set is changed from null to a non empty one
 	 */
 	@Test
-	public void setGroupMembers_shouldMarkTheObsAsDirtyWhenTheSetIsReplacedWithAnotherWithDifferentMembers()
-	    throws Exception {
+	public void setGroupMembers_shouldNotMarkNewObsAsDirtyWhenTheSetIsChangedFromNullToANonEmptyOne() throws Exception {
 		Obs obs = new Obs();
+		assertNull(Obs.class.getDeclaredField("groupMembers").get(obs));
+		Set members = new HashSet<>();
+		members.add(new Obs());
+		obs.setGroupMembers(members);
+		assertFalse(obs.isDirty());
+	}
+
+	/**
+	 * @see Obs#setGroupMembers(Set)
+	 * @verifies mark the existing obs as dirty when the set is replaced with another with different members
+	 */
+	@Test
+	public void setGroupMembers_shouldMarkTheExistingObsAsDirtyWhenTheSetIsReplacedWithAnotherWithDifferentMembers()
```

Comment:
Just like you did for most of the other tests, it would be nice to add a test like this that ensures a new Obs doesn't get marked as dirty